rcl/test/rcl/test_events.cpp

```diff
@@ -696,6 +696,103 @@ TEST_P(TestEventFixture, test_pubsub_incompatible_qos)
+    ret = rcl_event_fini(&subscription_event);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_subscription_fini(&subscription, this->node_ptr);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    return;
+  }
+
+  // (TODO blast545): Enable the rest of the test, timeout failing
+  // conditional_wait_for_msgs_and_events returns Timeout for rmw_cyclonedds_cpp
+  return;
+
+  WaitConditionPredicate events_ready = [](
+    const bool & /*msg_persist_ready*/,
+    const bool & subscription_persist_ready,
+    const bool & /*publisher_persist_ready*/) {
+      return subscription_persist_ready;
+    };
+  bool msg_persist_ready, subscription_persist_ready, publisher_persist_ready;
+  rcl_ret_t wait_res = conditional_wait_for_msgs_and_events(
+    context_ptr, MAX_WAIT_PER_TESTCASE, events_ready,
+    &subscription, &subscription_event, nullptr,
+    &msg_persist_ready, &subscription_persist_ready, &publisher_persist_ready);
+  EXPECT_EQ(wait_res, RCL_RET_OK);
+
+  // test that the subscriber/datareader discovered a lost message event
+  EXPECT_TRUE(subscription_persist_ready);
+  if (subscription_persist_ready) {
+    rmw_message_lost_status_t message_lost_status;
+    rcl_ret_t ret = rcl_take_event(&subscription_event, &message_lost_status);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    const auto & sub_total_count = message_lost_status.total_count;
+    const auto & sub_total_count_change = message_lost_status.total_count_change;
+    if (sub_total_count != 0 && sub_total_count_change != 0) {
+      EXPECT_EQ(sub_total_count, 1u) << error_msg;
+      EXPECT_EQ(sub_total_count_change, 1u) << error_msg;
+    } else {
+      ADD_FAILURE() << "Subscription incompatible message lost event timed out for: " << error_msg;
+    }
+  }
+
+  // clean up
+  ret = rcl_event_fini(&subscription_event);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  ret = rcl_subscription_fini(&subscription, this->node_ptr);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+}
+
+/*
+ * Passing bad params to functions not covered with other tests
+ */
+TEST_P(TestEventFixture, test_bad_args)
+{
+  setup_publisher_subscriber(default_qos_profile, default_qos_profile);
+  const rcl_subscription_event_type_t unknown_sub_type = (rcl_subscription_event_type_t) 5432;
+  const rcl_publisher_event_type_t unknown_pub_type = (rcl_publisher_event_type_t) 5432;
+
+  publisher_event = rcl_get_zero_initialized_event();
+  rcl_ret_t ret = rcl_publisher_event_init(
+    &publisher_event,
+    &publisher,
+    unknown_pub_type);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT);
+
+  subscription_event = rcl_get_zero_initialized_event();
+  ret = rcl_subscription_event_init(
+    &subscription_event,
+    &subscription,
+    unknown_sub_type);
+  EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT);
+
+  EXPECT_EQ(NULL, rcl_event_get_rmw_handle(nullptr));
+
+  tear_down_publisher_subscriber();
+}
```

Comment:
@Blast545 hmm, I'd rather not test unrelated API in the same test case.

Reply:
Done b0453b2